xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2423,8 +2444,11 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
             }
 
             // Look for a resource file
-            if (resourceExists("/resources/" + filename)) {
-                URL url = urlf.createResourceURL(filename, forceSkinAction, context);
+            String resourceFilePath = "/resources/" + filename;
+            if (resourceExists(resourceFilePath)) {
+                Map<String,String> queryParametersMap = getResourceURLCacheParameters(resourceFilePath);
+                queryParametersMap.putAll(queryParameters);
```

Comment:
So I updated the code to put the given queryParameters, in the only place I could use them and it actually feels wrong.
If you look at the stylesheets.vm I pushed in this same PR, we have the following line:
```
$urlParametersMap.put("colorTheme", $services.model.serialize($themeDoc.getDocumentReference(), "default")))
$xwiki.getSkinFile($stylesheet, true, ${urlParametersMap})
```

It means that basically those queryParameters will be discarded if we find the resource file in the current skin (previous lines of the same function) which doesn't feel right. I'd say that for this one I also need to add a method:

```
Resource#getURL(boolean, Map<String,String>)
```

to be able to ask a resource URL with some given parameters.

Reply:
After discussion we decided to remove the queryParameters argument in getSkinFile.